scipy/optimize/_linprog_highs.py

```diff
@@ -0,0 +1,207 @@
+"""HiGHS Linear Optimization Methods
+
+Interface to HiGHS linear optimization software.
+https://www.maths.ed.ac.uk/hall/HiGHS/
+
+.. versionadded:: 1.5.0
+
+References
+----------
+.. [1] Q. Huangfu and J.A.J. Hall. "Parallelizing the dual revised simplex
+           method." Mathematical Programming Computation, 10 (1), 119-142,
+           2018. DOI: 10.1007/s12532-017-0130-5
+
+"""
+
+import numpy as np
+from .optimize import _check_unknown_options
+from ._highs.highs_wrapper import highs_wrapper, CONST_INF
+from scipy.sparse import csc_matrix, vstack, issparse
+
+
+def _replace_inf(x):
+    # Replace `np.inf` with CONST_INF
+    infs = (np.abs(x) == np.inf)
+    x[infs] = np.sign(x[infs])*CONST_INF
+    return x
+
+
+def _linprog_highs(lp, solver, time_limit=1, presolve=True, parallel=False,
+                   disp=False, maxiter=None, autoscale=False,
+                   **unknown_options):
+    r"""
+    Solve the following linear programming problem using one of the HiGHS
+    solvers:
+
+    .. math::
+
+        \min_x \ & c^T x \\
+        \mbox{such that} \ & A_{ub} x \leq b_{ub},\\
+        & A_{eq} x = b_{eq},\\
+        & l \leq x \leq u ,
+
+    where :math:`x` is a vector of decision variables; :math:`c`,
+    :math:`b_{ub}`, :math:`b_{eq}`, :math:`l`, and :math:`u` are vectors; and
+    :math:`A_{ub}` and :math:`A_{eq}` are matrices.
+
+    Informally, that's:
+
+    minimize::
+
+        c @ x
+
+    such that::
+
+        A_ub @ x <= b_ub
+        A_eq @ x == b_eq
+        lb <= x <= ub
+
+    Note that by default ``lb = 0`` and ``ub = None`` unless specified with
+    ``bounds``.
+
+    Parameters
+    ----------
+    lp :  _LPProblem
+        A ``scipy.optimize._linprog_util._LPProblem`` ``namedtuple``.
+    solver : "ipm" or "simplex"
+        Which HiGHS solver to use.
```

Comment:
Maybe add "choose"

Reply:
Passing `None` will use the default `choose`